Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -27,6 +38,82 @@ void RegExpConstructor::initialize(Realm& realm)
+    define_native_accessor(realm, vm.names.lastParen, last_paren_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.lastParenAlias, last_paren_alias_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.leftContext, left_context_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.leftContextAlias, left_context_alias_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.rightContext, right_context_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.rightContextAlias, right_context_alias_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$1, group_1_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$2, group_2_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$3, group_3_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$4, group_4_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$5, group_5_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$6, group_6_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$7, group_7_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$8, group_8_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, vm.names.$9, group_9_getter, {}, Attribute::Configurable);
+}
+
+// UpdateLegacyRegExpStaticProperties ( C, S, startIndex, endIndex, capturedValues ) https://github.com/tc39/proposal-regexp-legacy-features#updatelegacyregexpstaticproperties--c-s-startindex-endindex-capturedvalues-
+void RegExpConstructor::update_legacy_static_properties(Utf16String const& string, size_t start_index, size_t end_index, Vector<Utf16View> const& captured_values)
+{
+    // 1. Assert: C is an Object that has a [[RegExpInput]] internal slot.
+    // 2. Assert: Type(S) is String.
+
+    // 3. Let len be the number of code units in S.
+    auto len = string.length_in_code_units();
+
+    // 4. Assert: startIndex and endIndex are integers such that 0 ≤ startIndex ≤ endIndex ≤ len.
+    VERIFY(start_index <= end_index);
+    VERIFY(end_index <= len);
+
+    // 5. Assert: capturedValues is a List of Strings.
+
+    // 6. Let n be the number of elements in capturedValues.
+    auto group_count = captured_values.size();
+
+    // 7. Set the value of C’s [[RegExpInput]] internal slot to S.
+    m_legacy_static_properties.set_input(string.view());
+
+    // 8. Set the value of C’s [[RegExpLastMatch]] internal slot to a String whose length is endIndex - startIndex and containing the code units from S with indices startIndex through endIndex - 1, in ascending order.
+    auto const& last_match = string.view().substring_view(start_index, end_index - start_index);
```

Comment:
`substring_view()` doesn't return a reference, so using one here is pointless

```suggestion
    auto last_match = string.view().substring_view(start_index, end_index - start_index);
```